src/main/java/slimeknights/tconstruct/library/recipe/alloy/inventory/IAlloyInventory.java

```diff
@@ -0,0 +1,18 @@
+package slimeknights.tconstruct.library.recipe.alloy.inventory;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraftforge.fluids.FluidStack;
+import slimeknights.tconstruct.library.recipe.fuel.IFluidInventory;
+
+import java.util.List;
+
+/**
+ * Inventory representing the contents of an alloy tank and surrounding tanks
+ */
+public interface IAlloyInventory extends IFluidInventory {
```

Comment:
I don't think `IFluidInventory` is a good choice. The fluid in that inventory was intended as a recipe input, we do not want allloy recipes to be tank output sensitive. Plus, the smeltery does not have just one fluid.

Make a method that checks if a fluid is valid for the tank instead. Smeltery always returns true, alloy tank will do fluid check there.

Reply:
Aren't we planning tiers of smelteries? Either way, we can do that and allow/disallow per tier.

Reply:
There is the potential that some smeltery variants may disallow certain fluids. For example, gasses. That will again be solved by a `allowsFluid(FluidStack)` method, much better than forcing all alloy inventories to conform to a single contained fluid.